### PR TITLE
Add minimum tags per topic setting.

### PR DESCRIPTION
**Pull request by yariplus**
request
https://community.nodebb.org/topic/5022/idea-the-minimum-number-of-tags

### CODE REVIEW

File: public/src/modules/composer/tags.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ define('composer/tags', function() {
 		}
 
 		tagEl.tagsinput({
+			minTags: config.minimumTagsPerTopic,
```
Comment (barisusakli) [inline public/src/modules/composer/tags.js]: I don't think bootstrap tagsinput has a minTags option

File: src/controllers/api.js [hunk 2]
```diff
@@ -55,6 +55,7 @@ apiController.getConfig = function(req, res, next) {
 	config.disableChat = parseInt(meta.config.disableChat, 10) === 1;
 	config.maxReconnectionAttempts = meta.config.maxReconnectionAttempts || 5;
 	config.reconnectionDelay = meta.config.reconnectionDelay || 1500;
+	config.minimumTagsPerTopic = meta.config.minimumTagsPerTopic || 0;
 	config.tagsPerTopic = meta.config.tagsPerTopic || 5;
```
Comment (barisusakli) [inline src/controllers/api.js]: tagsPerTopic should be renamed to maximumTagsPerTopic

File: src/topics/create.js [hunk 3]
```diff
@@ -290,6 +293,13 @@ module.exports = function(Topics) {
 		callback();
 	}
 
+	function checkTagsLength(tags, callback) {
+		if (!tags || tags.length < parseInt(meta.config.minimumTagsPerTopic, 10)) {
+			return callback(new Error('[[error:not-enough-tags, ' + meta.config.minimumTagsPerTopic + ']]'));
+		}
+		callback();
+	}
+
```
Comment (barisusakli) [inline src/topics/create.js]: Might as well check for tags.length > meta.config.maximumTagsPerTopic here as well.

**Comment by barisusakli** [comment]
The composer got moved to a plugin so some of the changes in this PR should go there I think.

**Comment by yariplus** [comment]
How's it look now?

and this
https://github.com/NodeBB/nodebb-plugin-composer-default/pull/1

File: src/socket.io/posts.js [hunk 4]
```diff
@@ -284,6 +284,8 @@ SocketPosts.edit = function(socket, data, callback) {
 		return callback(new Error('[[error:title-too-short, ' + meta.config.minimumTitleLength + ']]'));
 	} else if (data.title && data.title.length > parseInt(meta.config.maximumTitleLength, 10)) {
 		return callback(new Error('[[error:title-too-long, ' + meta.config.maximumTitleLength + ']]'));
+	} else if (data.title && data.tags && data.tags.length < parseInt(meta.config.minimumTagsPerTopic, 10)) {
```
Comment (barisusakli) [inline src/socket.io/posts.js]: No need to check data.title here

**Comment by barisusakli** [comment]
Also can you add [[error:not-enough-tags, %1]] and [[error:too-many-tags, %1]] into public/languages/en_GB/error.json ?

**Comment by yariplus** [comment]
:+1:

**Pull request merged**